### PR TITLE
run torch::deploy test using pip install

### PR DESCRIPTION
**Pull request by PaliC**
This PR runs the unit tests for [multipy](https://github.com/pytorch/multipy) in pytorch core such that we are able to make sure changes in core do not break multipy as adding `_prims` did.

Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #86507

### CODE REVIEW

**Review by anirbanr-fb-r2p — APPROVED** [review]
Lg in general, other than the cmake update steps.

File: .github/workflows/pull.yml [hunk 1]
```diff
@@ -172,6 +172,8 @@ jobs:
           { config: "distributed", shard: 2, num_shards: 3, runner: "linux.8xlarge.nvidia.gpu" },
           { config: "distributed", shard: 3, num_shards: 3, runner: "linux.8xlarge.nvidia.gpu" },
           { config: "functorch", shard: 1, num_shards: 1, runner: "linux.4xlarge.nvidia.gpu" },
+          { config: "deploy", shard: 1, num_shards: 1, runner: "linux.4xlarge.nvidia.gpu" },
+          { config: "deploy", shard: 1, num_shards: 1, runner: "linux.4xlarge.nvidia.gpu" },
```
Comment (d4l3k) [inline .github/workflows/pull.yml:176]: do we need GPUs for these tests? or is CPU sufficient?

File: .jenkins/pytorch/common_utils.sh [hunk 2]
```diff
@@ -137,6 +137,44 @@ function install_triton() {
...
 }
 
+function setup_torchdeploy_deps(){
+  conda install -y cmake
+  conda install -y -c conda-forge libpython-static=3.10
+  sudo wget -O - https://apt.kitware.com/keys/kitware-archive-latest.asc 2>/dev/null | sudo gpg --dearmor -o /usr/share/keyrings/magic-key.gpg && \
+  sudo echo "deb [arch=amd64,arm64 signed-by=/usr/share/keyrings/magic-key.gpg] https://apt.kitware.com/ubuntu/ bionic main" | sudo tee -a /etc/apt/sources.list && \
+  sudo echo "deb http://security.ubuntu.com/ubuntu focal-security main" | sudo tee -a /etc/apt/sources.list && \
+  sudo apt update && \
+  sudo apt install -y binutils && \
```
Comment (d4l3k) [inline .jenkins/pytorch/common_utils.sh:147]: after https://github.com/pytorch/multipy/pull/216 this shouldn't be required anymore

File: .jenkins/pytorch/common_utils.sh [hunk 3]
```diff
@@ -137,6 +137,44 @@ function install_triton() {
...
+  sudo rm -rf /var/lib/apt/lists/*
+  local CC="$(which gcc)"
+  local CXX="$(which g++)"
+  export CC
+  export CXX
```
Comment (d4l3k) [inline .jenkins/pytorch/common_utils.sh:152]: why `local` and not just `export CC`?

File: .jenkins/pytorch/common_utils.sh [hunk 4]
```diff
@@ -137,6 +137,44 @@ function install_triton() {
...
+  pip install --upgrade pip
+}
+
+function checkout_install_torchdeploy() {
+  local commit
+  setup_torchdeploy_deps
+  pushd ..
+  git clone https://github.com/pytorch/multipy
```
Comment (d4l3k) [inline .jenkins/pytorch/common_utils.sh:160]:
```
git clone --recurse-submodules https://github.com/pytorch/multipy.git
```

**Review by d4l3k — APPROVED** [review]
LGTM from multipy side -- lint is failing? Probably want someone from core to review as well

File: .github/workflows/pull.yml (hunk 1 above)
Comment (PaliC) [reply .github/workflows/pull.yml:176]: We only need 1 gpu (good catch on that). However, we are planning on adding gpu tests to do this right, for which we'd want the gpu. We talked about this offline.

File: .jenkins/pytorch/common_utils.sh (hunk 2 above)
Comment (PaliC) [reply .jenkins/pytorch/common_utils.sh:147]: will update

File: .jenkins/pytorch/common_utils.sh (hunk 3 above)
Comment (PaliC) [reply .jenkins/pytorch/common_utils.sh:152]: leads to a lint error :(

**Pull request closed without merging**